### PR TITLE
add cross-compiling support from project_options

### PR DESCRIPTION
**Pull request by abeimler**
https://github.com/aminya/setup-cpp/pull/103

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -50,6 +50,13 @@ tasks:
         FEATURE_TESTS: OFF
         CMAKE_BUILD_TYPE: Debug
 
+  cross_build_mingw:
+    - task: build_template
+      vars:
+        FEATURE_TESTS: OFF
+        CMAKE_BUILD_TYPE: Release
+        CONFIGURE_FLAGS: -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE="/home/cmake/toolchains/x86_64-w64-mingw32.toolchain.cmake" -DVCPKG_TARGET_TRIPLET="x64-mingw-dynamic"
```
Comment (aminya) [inline Taskfile.yml:58]: Is this path always correct?

**Review by aminya — CHANGES_REQUESTED** [review]

File: Taskfile.yml (hunk 1 above)
Comment (abeimler) [reply Taskfile.yml:58]: No, this just for testing in the Docker container.

Comment (aminya) [reply Taskfile.yml:58]: So I guess we should handle it in project options

https://github.com/aminya/project_options/issues/140#issuecomment-1283494365

**Comment by abeimler** [comment]
I would like to add [CPM](https://github.com/cpm-cmake/CPM.cmake) from [ModernCppStarter](https://github.com/TheLartians/ModernCppStarter) for downloading [project_options](https://github.com/aminya/project_options).

That would also open some future features like [Format.cmake](https://github.com/TheLartians/Format.cmake)

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,17 +4,29 @@ cmake_minimum_required(VERSION 3.16)
...
-FetchContent_Declare(_project_options URL https://github.com/aminya/project_options/archive/refs/tags/v0.26.2.zip)
-FetchContent_MakeAvailable(_project_options)
-include(${_project_options_SOURCE_DIR}/Index.cmake)
+# Change the version to update the package (watch the releases of the repository for future updates)
+include(cmake/CPM.cmake)
+CPMAddPackage(
+  NAME              project_options
+  GITHUB_REPOSITORY aminya/project_options
+  #VERSION           0.26.3
+  GIT_TAG           main
```
Comment (abeimler) [inline CMakeLists.txt:16]: Change this to a project_options version with `enable_cross_compiler()` ... _(Used this for testing)_

Comment (aminya) [reply CMakeLists.txt:16]: I prefer using `FetchContent`, which is built-in

**Review by aminya — CHANGES_REQUESTED** [review]

**Review by aminya — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -15,6 +15,10 @@ include(${_project_options_SOURCE_DIR}/Index.cmake)
 # Define the features of the project
 include("./Features.cmake")
 
+# enable cross-compiling: - should be called before run_vcpkg()
+if(ENABLE_CROSS_COMPILING)
+  enable_cross_compiler()
```
Comment (aminya) [inline CMakeLists.txt:20]: Does this need any parameters?

Comment (abeimler) [reply CMakeLists.txt:20]: > Does this need any parameters?
No, the fallback is Parameters (or env. variables). 

https://github.com/aminya/cpp_vcpkg_project/pull/15/files#diff-cd2d359855d0301ce190f1ec3b4c572ea690c83747f6df61c9340720e3d2425e
```yml
build_cross_mingw:
    - task: build_template
      vars:
        FEATURE_TESTS: OFF
        CMAKE_BUILD_TYPE: Release
        CONFIGURE_FLAGS: -DENABLE_CROSS_COMPILING:BOOL=ON -DDEFAULT_TRIPLET=x64-mingw-dynamic
```

Comment (aminya) [reply CMakeLists.txt:20]: Ok, thanks! I think we can add this to the documentation:
https://github.com/aminya/project_options/blob/main/docs/src/enable_cross_compiler.md

**Pull request merged**